### PR TITLE
Update to Go 1.22

### LINKED ISSUE #136: Go 1.22 upgrade check list
See https://github.com/nspcc-dev/neo-go/issues/3310

### PR DESCRIPTION
**Pull request by smallhive**
Closes #136

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ jobs:
     runs-on: ubuntu-22.04
```
Comment (roman-khimov) [inline .github/workflows/tests.yml:59]: Please update as well.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ This document outlines major changes between releases.
 
 ### Changed
 - Go 1.21+ is required to build now (#98)
+- Go 1.22+ is required to build now (#136)
```
Comment (roman-khimov) [inline CHANGELOG.md:9]: Can replace 1.21 above `...(#98, #136)`.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (smallhive) [reply .github/workflows/tests.yml]: This point should be added to https://github.com/nspcc-dev/neo-go/issues/3310 as well

File: CHANGELOG.md (hunk 2 above)
Comment (smallhive) [reply CHANGELOG.md]: Merged

File: .github/workflows/tests.yml (hunk 1 above)
Comment (roman-khimov) [reply .github/workflows/tests.yml]: Welcome to https://github.com/nspcc-dev/.github/issues/32

File: .github/workflows/builds.yml [hunk 3]
```diff
@@ -33,7 +33,7 @@ on:
 jobs:
   build_bin:
     name: Build binary
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
```
Comment (roman-khimov) [inline .github/workflows/builds.yml:36]: Separate commit and this changes the flow, btw. It was fixed and now it's latest.

Comment (smallhive) [reply .github/workflows/builds.yml:36]: Separated

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**